components/cam/src/physics/crm/samxx/vars.cpp

```diff
@@ -0,0 +1,1759 @@
+  yakl::memset(cloudtoptemp      ,0.);
+}
+
+
+void init_values() {
+  auto &z0   = ::z0  ;
+  auto &CF3D = ::CF3D;
+
+  flag_precip(0) = 0;
+  flag_precip(1) = 1;
+
+  dosubsidence = false    ;
+  ug = 0.                 ;
+  vg = 0.                 ;
+  les = false             ;
+  sfc_flx_fxd =false      ;
+  sfc_tau_fxd =false      ;
+  dodamping = false       ;
+  docloud = false         ;
+  docam_sfc_fluxes = false;
+  doprecip = false        ;
+  dosgs = false           ;
+  docoriolis = false      ;
+  dosurface = false       ;
+  dowallx = false         ;
+  dowally = false         ;
+  docolumn = false        ;
+  dotracers = false       ;
+  dosmoke = false         ;
+
+  parallel_for( ncrms , YAKL_LAMBDA ( int icrm ) {
+    z0(icrm) = 0.035;
+  });
+
+  na=1;
+  nb=2;
+  nc=3;
+
+  parallel_for( Bounds<4>(nzm,ny,nx,ncrms) , YAKL_LAMBDA (int k, int j, int i, int icrm) {
+    CF3D(k,j,i,icrm) = 1.;
+  });
+
+  nstep = 0                             ;
+  compute_reffc = false                 ;
+  compute_reffi = false                 ;
+  dx = 0.                               ;
+  dy = 0.                               ;
+  doconstdz = false                     ;
+  nstop =0                              ;
+  nelapse =999999999                    ;
+  dt=0.                                 ;
+  day0=0.                               ;
+  nrad =1                               ;
+  nrestart =0                           ;
+  nstat =1000                           ;
+  nstatfrq =50                          ;
+  restart_sep =false                    ;
+  nrestart_skip =0                      ;
+  output_sep =false                     ;
+  doisccp = false                       ;
+  domodis = false                       ;
+  domisr = false                        ;
+  dosimfilesout = false                 ;
+  doSAMconditionals = false             ;
+  dosatupdnconditionals = false         ;
+  doscamiopdata = false                 ;
+  dozero_out_day0 = false               ;
+  nsave3Dstart =99999999                ;
+  nsave3Dend  =99999999                 ;
+  save3Dbin =false                      ;
+  save3Dsep =false                      ;
+  qnsave3D =0.                          ;
+  dogzip3D =false                       ;
+  rad3Dout = false                      ;
+  nsave2D =1000                         ;
+  nsave2Dstart =99999999                ;
+  nsave2Dend =99999999                  ;
+  save2Dbin =false                      ;
+  save2Dsep =false                      ;
+  save2Davg =false                      ;
+  dogzip2D =false                       ;
+  nstatmom =1000                        ;
+  nstatmomstart =99999999               ;
+  nstatmomend =99999999                 ;
+  savemomsep =false                     ;
+  savemombin =false                     ;
+  nmovie =1000                          ;
+  nmoviestart =99999999                 ;
+  nmovieend =99999999                   ;
+  isInitialized_scamiopdata = false     ;
```

Comment:
A lot of these variables aren't used here. They were just carried over from the standalone SAM and are menat to control SAM's output and restart files. So we could delete some or all. We'd just need to be careful to check that they aren't used, but I'm 90% sure they are just set and no used.